src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -217,13 +218,23 @@ protected Future<Boolean> receiveInventoryItemAndUpdatePiece(JsonObject item, Pi
       })
       // Add processing error if item failed to be updated
       .onFailure(e -> {
-        logger.error("Item associated with piece '{}' cannot be updated", piece.getId());
-        addError(piece.getPoLineId(), piece.getId(), ITEM_UPDATE_FAILED.toError());
```

Comment:
this invocation of addError method() has 5 occurences with code ITEM_UPDATE_FAILED but only 2 of them are matches with ITEM_UPDATE_FAILED: CheckingHelper and ReceivingHelper.
All occurences in abstract class CheckingReceivePiecesHelper should not use ITEM_UPDATE_FAILED, they should use error codes like po line not found, title not foud, hodling can not be created - but we can improve this in next story.
For this story propose to move new function in base CheckinReceivePiecesHelper and use it in 2 places where ITEM_UPDATE_FAILED error code makes sense(when actualy dealing with item update): CheckingHelper and ReceivingHelper.
![image](https://github.com/folio-org/mod-orders/assets/25097693/aa0c9929-b5cb-4830-b91a-ba963c57af34)

Reply:
or you can do this refactoring to use correct error codes in CheckingReceivePiecesHelper in scope of this story, in this case we will notify UI devs that we have new error codes

Reply:
Thank you, Serhii, for point that out! will do